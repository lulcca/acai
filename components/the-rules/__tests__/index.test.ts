import { beforeEach, describe, expect, it, vi } from 'vitest';
import TheRules from '@/components/the-rules/index.vue';
import { mount } from '@vue/test-utils';

describe('Components > TheRules', () => {
  vi.mock('assets/json/rules.json', () => ({
    default: ['Regra 1', 'Regra 2', 'Regra 3'],
  }));

  let wrapper: ReturnType<typeof mount>;

  beforeEach(() => wrapper = mount(TheRules));

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the title "Regras"', () => {
    expect(wrapper.find('h2').text()).toBe('Regras');
  });

  it('should render the rules list correctly', () => {
    const listItems = wrapper.findAll('li');
    expect(listItems).toHaveLength(3);
    expect(listItems[0].text()).toBe('Regra 1');
    expect(listItems[1].text()).toBe('Regra 2');
    expect(listItems[2].text()).toBe('Regra 3');
  });
});
