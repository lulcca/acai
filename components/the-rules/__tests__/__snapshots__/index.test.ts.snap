// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > TheRules > should match snapshot 1`] = `
"<section id="regras" class="gap-10 grid grid-cols-1 grid-rows-[auto_1fr] h-screen place-content-start place-items-start w-full">
  <h2 class="font-major text-6xl uppercase"> Regras </h2>
  <ol class="list-decimal">
    <li class="text-lg">Regra 1</li>
    <li class="text-lg">Regra 2</li>
    <li class="text-lg">Regra 3</li>
  </ol>
</section>"
`;
